src/nlua_misn.c

```diff
@@ -109,6 +112,8 @@ static const luaL_Reg misn_methods[] = {
    { "npcAdd", misn_npcAdd },
    { "npcRm", misn_npcRm },
    { "claim", misn_claim },
+   { "setLogID", misn_setLogID },
+   { "getLogID", misn_getLogID },
```

Comment:
Are these needed for something? I see that you used them in the Shadow Vigil logs you specified but they seem to be unnecessary, i.e. it seems simple variables would do just as well.

🕵️

Reply:
Yes, required to set/get the logid for the mission, which will ensure that it is saved correctly.

Reply:
Oh, is it because it can change between saves? Hm... it would be better for that to be transparently handled in the background (as it is for pilots for example). Would there be any way to do that, so that you could just store the log ID or whatever in a variable and rely on it continuing to work through saves? That's how we do things in the rest of the game, requiring this different way is confusing and prone to errors.

Alternatively it might be better to merge "creating" the log and "appending" the log into one function? So that it would just look up the proper log by name at the appropriate time. Basically you would just have appendLog include the identifier string as an argument instead of log ID and it would find out the log ID purely through C code (and not expose that to the Lua scripter).

Anyway, just some thoughts. If no change is feasible this quirk at least needs to be documented.

🕵️